packages/frontend-shared/cypress/e2e/e2ePluginSetup.ts

```diff
@@ -155,42 +157,57 @@ async function makeE2ETasks () {
 
       sinon.stub(ctx.util, 'fetch').get(() => {
         return async (url: RequestInfo, init?: RequestInit) => {
-          if (!String(url).endsWith('/test-runner-graphql')) {
-            return fetchApi(url, init)
+          if (String(url).endsWith('/test-runner-graphql')) {
+            const { query, variables } = JSON.parse(String(init?.body))
+            const document = parse(query)
+            const operationName = getOperationName(document)
+
+            let result = await execute({
+              operationName,
+              document,
+              variableValues: variables,
+              schema: cloudSchema,
+              rootValue: CloudRunQuery,
+              contextValue: {
+                __server__: ctx,
+              },
+            })
+
+            if (remoteGraphQLIntercept) {
+              try {
+                result = await remoteGraphQLIntercept({
+                  operationName,
+                  variables,
+                  document,
+                  query,
+                  result,
+                })
+              } catch (e) {
+                const err = e as Error
+
+                result = { data: null, extensions: [], errors: [new GraphQLError(err.message, undefined, undefined, undefined, undefined, err)] }
+              }
+            }
+
+            return new Response(JSON.stringify(result), { status: 200 })
           }
 
-          const { query, variables } = JSON.parse(String(init?.body))
-          const document = parse(query)
-          const operationName = getOperationName(document)
-
-          let result = await execute({
-            operationName,
-            document,
-            variableValues: variables,
-            schema: cloudSchema,
-            rootValue: CloudRunQuery,
-            contextValue: {
-              __server__: ctx,
-            },
-          })
-
-          if (remoteGraphQLIntercept) {
-            try {
-              result = await remoteGraphQLIntercept({
-                operationName,
-                variables,
-                document,
-                query,
-                result,
-              })
-            } catch (e) {
-              const err = e as Error
-
-              result = { data: null, extensions: [], errors: [new GraphQLError(err.message, undefined, undefined, undefined, undefined, err)] }
-            }
+          if (String(url) === 'https://download.cypress.io/desktop.json') {
+            return new Response(JSON.stringify({
+              name: 'Cypress',
+              version: pkg.version,
+            }), { status: 200 })
+          }
+
+          if (String(url) === 'https://registry.npmjs.org/cypress') {
+            return new Response(JSON.stringify({
+              'time': {
+                [pkg.version]: '2022-02-10T01:07:37.369Z',
+              },
+            }), { status: 200 })
```

Comment:
I'm not sure if supertest is quite the right fit here since this is more just stubbing out the responses for the e2e, but it might be worth having a "stub" express server we can setup for our tests for anything hitting external apis.